cc_utilities/redcap_sync.py

```diff
@@ -111,22 +121,199 @@ def set_external_id_column(df, external_id_col):
     For the given external_id_col, drop any rows with no value and
     copy to a new column named "external_id"
     """
+    df = df.copy()
     df = df.dropna(subset=[external_id_col])
-    df.loc[:, "external_id"] = df[external_id_col]
+    df["external_id"] = df[external_id_col]
     return df
 
 
-def split_complete_and_incomplete_records(cases_df):
+def get_external_ids_and_dobs(
```

Comment:
This looks good and works for me! 🎉 

Optional / if desired: `read_sql` could be employed to shorten the function a little:

```
return pd.read_sql(
    f"""SELECT
            {external_id_col},
            {DOB_FIELD}
        FROM {table_name}
        WHERE
            {external_id_col} IN %(external_ids)s
            AND {DOB_FIELD} IS NOT NULL
            AND {DOB_FIELD} <> ''
    """,
    create_engine(db_url),
    params={"external_ids": tuple(external_ids)}
).to_dict(orient="records")
```

Reply:
:sunglasses: